pinot-controller/src/main/java/com/linkedin/pinot/controller/api/upload/ZKOperator.java

```diff
@@ -157,9 +158,10 @@ private void processExistingSegment(SegmentMetadata segmentMetadata, URI finalSe
         LOGGER.info(
             "New segment crc {} is different than the existing segment crc {}. Updating ZK metadata and refreshing segment {}",
             newCrc, existingCrc, segmentName);
-        moveSegmentToPermanentDirectory(currentSegmentLocation, finalSegmentLocationURI);
-        LOGGER.info("Moved segment {} from temp location {} to {}", segmentName,
-            currentSegmentLocation.getAbsolutePath(), finalSegmentLocationURI.getPath());
+        if (moveSegmentToFinalLocation) {
+          moveSegmentToPermanentDirectory(currentSegmentLocation, finalSegmentLocationURI);
+          LOGGER.info("Moved segment {} from temp location {} to {}", segmentName, currentSegmentLocation.getAbsolutePath(), finalSegmentLocationURI.getPath());
+        }
```

Comment:
may be useful to add a log here in an else that we are skipping move to final location